src/main/java/lotto/domain/LottoNumber.java

```diff
@@ -1,23 +1,37 @@
 package lotto.domain;
 
+import java.util.HashMap;
+import java.util.Map;
 import java.util.Objects;
 
 public class LottoNumber {
     private static final int MIN_NUMBER = 1;
     private static final int MAX_NUMBER = 45;
+    private static final Map<Integer, LottoNumber> CACHE = new HashMap<>();
+
+    static {
+        for (int i = MIN_NUMBER; i <= MAX_NUMBER; i++) {
+            CACHE.put(i, new LottoNumber(i));
+        }
+    }
 
     private final int value;
 
-    public LottoNumber(String value) {
+    private LottoNumber(String value) {
         this(Integer.parseInt(value));
     }
```

Comment:
정적 팩토리 메서드를 사용할 경우 생성자는 기본 생성자 하나로 충분
값의 변환은 정적 팩터리 메서드가 처리
```suggestion
    public static LottoNumber of(String number) {
        return CACHE.get(Integer.parseInt(number));
    }
```